crates/oxc_cli/src/lint/runner_with_module_tree.rs

```diff
@@ -0,0 +1,324 @@
+                visited: Arc::clone(&visited),
+                tx_error,
+            },
+        );
+
+        let (number_of_warnings, number_of_diagnostics) = self.process_diagnostics(&rx_error);
+
+        CliRunResult::LintResult {
+            duration: now.elapsed(),
+            number_of_rules: linter.number_of_rules(),
+            number_of_files: visited.len(),
+            number_of_diagnostics,
+            number_of_warnings,
+            max_warnings_exceeded: self
+                .options
+                .max_warnings
+                .map_or(false, |max_warnings| number_of_warnings > max_warnings),
+        }
+    }
+
+    fn process_diagnostics(&self, rx_error: &Receiver<(PathBuf, Vec<Error>)>) -> (usize, usize) {
+        let mut number_of_warnings = 0;
+        let mut number_of_diagnostics = 0;
+        let mut buf_writer = BufWriter::new(std::io::stdout());
+        let handler = GraphicalReportHandler::new();
+
+        for (path, diagnostics) in rx_error.iter() {
+            number_of_diagnostics += diagnostics.len();
+
+            let mut output = String::new();
+
+            for diagnostic in diagnostics {
+                if diagnostic.severity() == Some(Severity::Warning) {
+                    number_of_warnings += 1;
+                    // The --quiet flag follows ESLint's --quiet behavior as documented here: https://eslint.org/docs/latest/use/command-line-interface#--quiet
+                    // Note that it does not disable ALL diagnostics, only Warning diagnostics
+                    if self.options.quiet {
+                        continue;
+                    }
+
+                    if let Some(max_warnings) = self.options.max_warnings {
+                        if number_of_warnings > max_warnings {
+                            continue;
+                        }
+                    }
+                }
+
+                let mut err = String::new();
+                handler.render_report(&mut err, diagnostic.as_ref()).unwrap();
+
+                if err.lines().all(|line| line.len() < 400) {
+                    output.push_str(&err);
+                    continue;
+                }
+
+                // If the error is too long, we assume it's a minified file
+                let minified_diagnostic = Error::new(MinifiedFileError(path.clone()));
+                output = format!("{minified_diagnostic:?}");
+                break;
+            }
+
+            buf_writer.write_all(output.as_bytes()).unwrap();
+        }
+
+        buf_writer.flush().unwrap();
+        (number_of_warnings, number_of_diagnostics)
+    }
+}
+
+fn process_paths(paths: &[PathBuf], runtime_data: LinterRuntimeData) {
+    paths.par_iter().for_each(move |path| {
+        let path = path.canonicalize().unwrap();
+
+        if path.is_file() {
+            run_for_file(&path, &runtime_data);
+        } else if path.is_dir() {
+            run_for_dir(&path, &runtime_data);
+        }
```

Comment:
Symbolic links should be handled to avoid recursive traversals

Reply:
Aren't they already handled by the `.canonicalize()`? And later by `nodejs_resolver`

Reply:
Yes the path is canonicalized by nodejs_resolver